upstream/gql.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package upstream
+
+import (
+	"context"
+
+	"go.mondoo.com/cnquery/v9/providers-sdk/v1/upstream/gql"
+	policy "go.mondoo.com/cnspec/v9/policy"
+	mondoogql "go.mondoo.com/mondoo-go"
+)
+
+type PageInfo struct {
+	StartCursor     string `json:"startCursor"`
+	EndCursor       string `json:"endCursor"`
+	HasNextPage     bool   `json:"hasNextPage"`
+	HasPreviousPage bool   `json:"hasPreviousPage"`
+}
+
+func SearchPolicy(
+	ctx context.Context,
+	c *gql.MondooClient,
+	scopeMrn string,
+	assingedOnly,
+	includePublic,
+	includePrivate *bool,
+) ([]*policy.Policy, error) {
+	var q struct {
+		Content struct {
+			TotalCount int `json:"totalCount"`
+			Edges      []struct {
+				Cursor string `json:"cursor"`
+				Node   struct {
+					Policy struct {
+						Mrn  string
+						Name string
```

Comment:
We should fetch the `action` here as well, so that we know the policy is enabled, preview not disabled. In addition, `trustLevel` and `access` help us to determine if the policy is private and uploaded to the space.